File: tests/fixtures/kotlin/Simple.kt
/**
 * FIXTURE: Simple Kotlin class
 * TESTS: Basic function/class extraction
 */

package com.example

import java.util.UUID

data class User(val id: UUID, val name: String, val email: String)

interface UserRepository {
    fun findById(id: UUID): User?
    fun save(user: User): User
    fun delete(id: UUID)
}

class UserService(private val repository: UserRepository) {
    fun getUser(id: UUID): User {
        return repository.findById(id) ?: throw NoSuchElementException("User not found")
    }

    suspend fun createUser(name: String, email: String): User {
        val user = User(UUID.randomUUID(), name, email)
        return repository.save(user)
    }

    fun deleteUser(id: UUID) {
        repository.delete(id)
    }
}

fun add(a: Int, b: Int): Int = a + b

File: tests/fixtures/swift/Simple.swift
/**
 * FIXTURE: Simple Swift struct and protocol
 * TESTS: Basic function/struct extraction
 */

import Foundation

struct User {
    let id: UUID
    let name: String
    let email: String
}

protocol UserRepository {
    func findById(_ id: UUID) -> User?
    func save(_ user: User) -> User
    func delete(_ id: UUID)
}

class UserService {
    private let repository: UserRepository

    init(repository: UserRepository) {
        self.repository = repository
    }

    func getUser(id: UUID) -> User {
        guard let user = repository.findById(id) else {
            fatalError("User not found")
        }
        return user
    }

    func deleteUser(id: UUID) {
        repository.delete(id)
    }
}

func add(_ a: Int, _ b: Int) -> Int {
    return a + b
}
